### ISSUE TITLE
`no_extension_access_modifier` rule triggers incorrectly for `nonisolated extension`

### ISSUE THREAD

**Issue by lukaskollmer** [realm/SwiftLint#6168]
### Bug Description

The `no_extension_access_modifier` rule triggers for a `nonisolated extension`, which is incorrect since the `nonisolated` here isn't an access control modifier.
(The ability to use the `nonisolated` keyword on extensions was [added with Swift 6.1](https://www.swift.org/blog/swift-6.1-released/).)

```swift
@MainActor
struct Test {}

nonisolated extension Test {} // No Extension Access Modifier Violation: Prefer not to use extension access modifiers (no_extension_access_modifier)
```

### Environment

* SwiftLint version: 0.59.1
* Xcode version: Xcode 16.4 (16F6)
* Installation method used: built from source
* Configuration file:
    ```yml
    enabled_rules:
      - no_extension_access_modifier
    ```
- no nested configurations
